sklearn/decomposition/dict_learning.py

```diff
@@ -110,7 +110,8 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         clf = Lasso(alpha=alpha, fit_intercept=False, precompute=gram,
                     max_iter=max_iter, warm_start=True)
         clf.coef_ = init
-        clf.fit(dictionary.T, X.T)
+        # Copying X to avoid buffer source-array read only error
+        clf.fit(dictionary.T, X.copy().T)
```

Comment:
I think it is related to the way joblib handles shared array. It has some common points with  #4597.
I added an example in issue #4772